lib/open_food_network/permalink_generator.rb

```diff
@@ -16,7 +23,23 @@ def find_available_value(existing, requested)
   end
 
   def create_unique_permalink(requested)
-    existing = self.class.where('id != ?', id).where("permalink LIKE ?", "#{requested}%").pluck(:permalink)
+    existing = others.where("permalink LIKE ?", "#{requested}%").pluck(:permalink)
```

Comment:
this is a module included in the product model but we are a bit far from the product model here, I'd add a bit more context to this method name by calling it other_products instead of others.

Reply:
Careful, this class is also used in `EnterpriseGroup`.

Instead of heaving an OFN specific service, I would rather see an independent module that can live in lib or in a separate gem. And that's where we could move to Friendly Id at some point.

Another solution would be to overwrite permalinks when deleting products so that the permalink is available to new products again. Spree did that but I'm not sure we want to re-use permalinks. We don't actually use them for products at the moment. Only for enterprises and groups, I think.

Reply:
ah, I see.
Friendly Id looks good!